### PR TITLE
parse: remaining per-collector goldens (T044, T045, T063, T064, T066, T067)

### PR DESCRIPTION
**Pull request by matias-sanchez**
## Summary

Third PR of the parser-goldens bucket. Closes the **parser-side of Principle VIII coverage** for every pt-stalk collector My-gather consumes by adding six new tests and four committed goldens — one per remaining collector (top, vmstat, innodbstatus, mysqladmin) plus two mysqladmin edge-case tests (snapshot-boundary reset, variable drift).

After this PR, `testdata/golden/` holds seven parser goldens covering every per-collector output (iostat from PR #5, processlist + variables from PR #4, plus top/vmstat/innodbstatus/mysqladmin here). Remaining golden work is render-HTML goldens (T046, T057, T069, T096) — separate bucket.

## Tasks closed

| Task | Test | Fixture → Golden |
|---|---|---|
| **T044** (FR-010) | `parse/top_test.go::TestTopGolden` | `-top` → `top.example2.2026_04_21_16_51_41.json` |
| **T045** (FR-011) | `parse/vmstat_test.go::TestVmstatGolden` | `-vmstat` → `vmstat.example2.2026_04_21_16_51_41.json` |
| **T063** (FR-014) | `parse/innodbstatus_test.go::TestInnoDBStatusGolden` | `-innodbstatus1` → `innodbstatus.example2.2026_04_21_16_51_41.json` |
| **T064** (FR-028) | `parse/mysqladmin_test.go::TestMysqladminGolden` | `-mysqladmin` → `mysqladmin.example2.2026_04_21_16_51_41.json` |
| **T066** (FR-030) | `parse/mysqladmin_test.go::TestSnapshotBoundaryReset` | (synthesised via internal parser API) |
| **T067** (R8.C) | `parse/mysqladmin_test.go::TestVariableDrift` | (in-memory synthetic blocks) |

### `goldens.ScrubFloats` first consumer

PR #4 landed `ScrubFloats` as an opt-in NaN-scrubber for parsers that deliberately emit `math.NaN()` — `parse/mysqladmin` is the only such parser in the MVP. `TestMysqladminGolden` routes `data.Deltas` through `ScrubFloats` before marshalling; the committed golden contains one `null` slot (drift pad at the end of a variable whose last samples were absent in the committed fixture), exercising the NaN path end-to-end.

### Top ranking: the F7 + data-model.md subtlety

FR-010 (via F7 remediation) says Top3ByAverage ranks by `sum / totalBatchCount`, absent-in-batch counts as 0. `data-model.md` additionally says `ProcessSeries.CPU.Samples` contains only *observed* samples (not zero-padded to `totalBatchCount`). The parser implements both: divides by `totalBatchCount` for the ranking, stores observed-only samples. `TestTopGolden` pins **both** halves with inline assertions and recomputes the ranking denominator from `distinctTimestamps`, not from `len(Samples)`, so a refactor can't silently flip either semantic.

### Mysqladmin edge cases

T066 and T067 use the internal `newMysqladminParser` + `parseOneFile` + `ResetForNewSnapshot` + `finish` API (the same API `render/`'s concat code calls). `parseMysqladmin` the public function is a single-file convenience wrapper and can't produce boundary behaviour on its own.

T067 uses **in-memory synthetic blocks** rather than a committed fixture so the drift condition is deterministic and independent of any fixture refresh: the first block reports three variables, the second block drops one. A committed "drift fixture" would bind this test to fixture surgery forever; inline blocks keep the assertion readable.

## Files touched

| File | Change |
|---|---|
| `parse/top_test.go` | **NEW** — `TestTopGolden` (T044). |
| `parse/vmstat_test.go` | **NEW** — `TestVmstatGolden` (T045). |
| `parse/innodbstatus_test.go` | **NEW** — `TestInnoDBStatusGolden` (T063). |
| `parse/mysqladmin_test.go` | Extended — `TestMysqladminGolden` (T064), `TestSnapshotBoundaryReset` (T066), `TestVariableDrift` (T067). Existing `TestPtMextFixture` (T065) untouched. |
| `testdata/golden/top.example2.2026_04_21_16_51_41.json` | **NEW** — 53 KB. |
| `testdata/golden/vmstat.example2.2026_04_21_16_51_41.json` | **NEW** — 54 KB. |
| `testdata/golden/innodbstatus.example2.2026_04_21_16_51_41.json` | **NEW** — 274 B (scalar record). |
| `testdata/golden/mysqladmin.example2.2026_04_21_16_51_41.json` | **NEW** — 217 KB (80+ counters × 30 samples, one `null` slot from drift). |
| `specs/001-ptstalk-report-mvp/tasks.md` | T044/T045/T063/T064/T066/T067 marked `[x]`; Reconciliation Summary 68 → **74 / 119** (≈62 %). |

## Regenerate goldens

```bash
go test ./parse/... -update
git diff testdata/golden/
```

The `-update` flag is registered only in packages that import `tests/goldens`; see the package godoc for the scoping rule introduced in PR #4's round-2 fixes.

## Verification

- `go test ./...` green on the full tree.
- `go test ./parse/... -update` is idempotent — second run produces zero diff in `testdata/golden/` (determinism).
- `goldens.ScrubFloats` first real consumer confirmed working end-to-end (one `null` in mysqladmin golden).
- No new Go dependencies; no production code touched.

## Review guidance

Same rules as every prior PR in this feature:

- **Active development, NOT production.** Do NOT suggest duplicated code, fallbacks, compatibility shims, defensive validation for impossible cases, or speculative abstractions.
- Every suggestion should **reduce code, sharpen an invariant, or close a real drift**.
- Specifically welcome:
  - sharper invariants to assert inline (e.g., counter arithmetic post-boundary, not just NaN-at-boundary);
  - any shape in a golden that looks fragile to a later parser change;
  - concerns about the size of `mysqladmin.example2.*.json` (217 KB) — I believe it's defensible (the fixture has 80+ counters × 30 samples × NaN-preserving JSON) but open to alternative framings.

## Test plan

- [ ] `go test ./parse/... -run 'TestTopGolden|TestVmstatGolden|TestInnoDBStatusGolden|TestMysqladminGolden|TestSnapshotBoundaryReset|TestVariableDrift' -v` passes without `-update`.
- [ ] `git diff testdata/golden/` is empty after a fresh `go test ./parse/... -update` run (idempotent).
- [ ] `tasks.md` Reconciliation Summary shows 74 / 119.
- [ ] CI green on the full chain.

## Next buckets after this merges

Per the bucket plan (6 PRs instead of 51 tasks):

- **PR B** — render HTML goldens (T046, T057, T058, T069, T070, T096), depends on this PR.
- **PR C** — FR-033..037 feature tests (T098, T100, T102, T104, T106), depends on PR B.
- **PR D** — CLI + integration (T030-T035, T086, T088, T091-T095).
- **PR E** — fixtures + coverage + recovery extra (T019, T080, T089, T090).
- **PR F** — perf + memory + quality (T078, T079, T081, T082, T085).

Separate work:

- CSS refactor (T108 + T109) — its own PR.
- Phase 8 audit (T110-T119) — manual work.

https://claude.ai/code/session_01EzEaXYpKAsnJqyef4wydMP

### CODE REVIEW

File: specs/001-ptstalk-report-mvp/tasks.md [hunk 1]
```diff
@@ -389,23 +389,23 @@ A mechanical pass compared every task ID against the real repository state for t
...
 
 **Fixture / golden gaps**:
 
 - **T019**: `testdata/example1/` was never committed — only `testdata/example2/` exists. Any test that keys off a distinct first example (or that asserts two format variants co-exist per FR-024) depends on this.
-- `testdata/golden/` is **partially populated**: T056 (variables) and T068 (processlist) goldens are committed as of this PR. Remaining `*Golden*` tests (T043, T044, T045, T046, T057, T063, T064, T069) still need committed goldens under `testdata/golden/` generated via `go test ./parse/... -update` and reviewed. Until those are added, those tests either fail or skip.
+- `testdata/golden/` holds **seven parser goldens** — one per collector: iostat, top, vmstat, innodbstatus, mysqladmin, processlist, variables. The remaining `*Golden*` tests are render-HTML goldens (T046, T057, T069, T096) which still need committed fixtures under `testdata/golden/` generated via `go test ./render/... -update` and reviewed. Parser-side golden coverage is complete.
```
[the remark this answers is not in the document]
Comment (matias-sanchez) [reply specs/001-ptstalk-report-mvp/tasks.md]: Addressed by `c39e2e7` (G5). Fair point — `render/` doesn't import `tests/goldens` today, so `-update` is unregistered in render test binaries. Qualified the instruction per your suggestion: the `go test ./render/... -update` command becomes correct only once render tests import `tests/goldens` (or otherwise wire up the scoped flag); until then the regen command should be scoped to the package(s) that actually register it. Pointed at the package godoc's "Scope of the -update flag" note so the rule stays discoverable.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: specs/001-ptstalk-report-mvp/tasks.md [hunk 2]
```diff
@@ -128,9 +128,9 @@ All tasks write paths relative to the repository root.
...
+- [x] T044 [P] [US2] `parse/top_test.go::TestTopGolden`: parse `-top` fixture; compare against golden. Also asserts `Top3ByAverage` is ranked with **average CPUPercent normalised over total batch count** (absent-in-batch counts as 0 — F7 resolution, aligning with spec FR-010 "aggregate"). The (higher PID first) tiebreaker is pinned by a deterministic synthetic-tie assertion inside the test (floating-point averages from a real fixture rarely compare equal, so a fixture-only check would leave the tiebreaker branch effectively unexercised). Golden at `testdata/golden/top.example2.2026_04_21_16_51_41.json`; regenerate with `go test ./parse/... -update`.
+- [x] T045 [P] [US2] `parse/vmstat_test.go::TestVmstatGolden`: parse `-vmstat` fixture; compare against golden. Asserts the declared `Series` order is present (runqueue, blocked, free_kb, buff_kb, cache_kb, swap_in, swap_out, io_in, io_out, cpu_user, cpu_sys, cpu_idle, cpu_iowait) and that every non-empty Series shares the same sample count. Golden at `testdata/golden/vmstat.example2.2026_04_21_16_51_41.json`.
```
[the remark this answers is not in the document]
Comment (matias-sanchez) [reply specs/001-ptstalk-report-mvp/tasks.md]: Addressed by `721449d` (N2). T050 was the real factual error — it still claimed `parse/vmstat.go` emits "diagnostics for missing columns" across the two format variants, which the parser simply doesn't do (zero-fills silently per M3/M4). Removed the false claim and replaced with the real silent-zero-fill behaviour. T045's description already captures the test's coverage; T050 now describes the implementation accurately. Good eye for the cross-section drift.

---
_Generated by [Claude Code](https://claude.ai/code)_
